### PR TITLE
feat: updated table UX, additional 'all' rows option

### LINKED ISSUE #1280: ⚡️ Feature: Container View - Checkboxes and Collapsed Groups
### Feature Description

Two features in one request, but same page so I figured I'd leave it as I know lots of issues/reqs can get hard to manage.

I love in Portainer how I can select a group of containers and delete them, often a stack, I see no row of checkboxes in the container UI. As well, I LOVE the group/stack view, but it should be collapsed by default. 

Finally, I'd like to see an "All" option for rows/page like Portainer.

Beautiful UI! But needs some more work, will be watching this project for when I can make the switch!

PS: Bug 1274 Re: Crashing when Agent unavailable is a big one to me as well, I have Portainer setup to view cloud VMs that are often unavailable and I accidentally click, it needs to handle that gracefully.

### Problem It Solves

Better UI View and ability to group action on containers.

### Proposed Solution

Read Top

### Alternatives Considered

_No response_

### Additional Context

_No response_

### Would you be willing to work on this feature?

- [ ] Yes, I'd like to implement this feature
- [ ] I could help with parts of this feature
- [ ] No, I'm just suggesting the feature

### LINKED ISSUE #1363: ⚡️ Feature: table-based view of project's container
### Feature Description

I would like to have a table-based view displaying the containers of a project.

### Problem It Solves

When a project contains multiple containers, card-based view makes it harder to quickly compare their states and properties. Important information such as status, image, ports, resource usage, or uptime is not easily scannable or comparable across containers.

### Proposed Solution

Provide a table view for project containers, where each container is displayed as a row and key attributes are shown as columns (for example: name, status, image, ports, CPU/memory usage, uptime), and controls. The table should support basic interactions such as sorting, filtering, and column visibility to adapt to different workflows.

### Alternatives Considered

The current list or card view can still be used, but it is less efficient for projects with many containers. Switching between containers one by one does not provide the same level of overview as a tabular layout.

### Additional Context

_No response_

### Would you be willing to work on this feature?

- [ ] Yes, I'd like to implement this feature
- [ ] I could help with parts of this feature
- [ ] No, I'm just suggesting the feature

### PR DESCRIPTION
**Pull request by cabaucom376**
Closes #1280
Closes #1363

- [x] Table grouping w/ bulk selection
- [x] Various missing info on mobile cards
- [x] Table groupings collapsed by default
- [x] Update filtering for containers table
- [x] "All" pagination filter option
- [x] Project containers table

Bonus:
- [x] Slimmer more visually appealing app-wide scrollbar :)

image

**Disclaimer** Greptiles Reviews use AI, make sure to check over its work.

To better help train Greptile on our codebase, if the comment is useful and valid **Like** the comment, if its not helpful or invalid **Dislike**

<h2>Greptile Overview</h2>

<details open><summary><h3>Greptile Summary</h3></summary>

This PR implements major table UX improvements with grouping, bulk selection, and an "All" pagination option.

**Key Changes:**
- **Backend**: Added `limit=-1` support for "show all" pagination across all services (containers, images, networks, volumes, projects, templates), with comprehensive test coverage
- **Frontend**: Enhanced table component with collapsible grouping, bulk selection within groups, sticky columns, and improved mobile cards
- **New Features**: 
  - Project containers table with service-level bulk actions
  - Update status filtering for containers (has_update, up_to_date, error, unknown)
  - Custom scrollbar styling that appears on hover
- **Code Quality**: Refactored duplicate pagination logic into reusable `BuildResponseFromFilterResult` helper, consistent across all services

The implementation is well-tested, follows existing patterns, and delivers the requested features (closes #1280 and #1363).
</details>

<details open><summary><h3>Confidence Score: 5/5</h3></summary>

- This PR is safe to merge with minimal risk
- The implementation is thorough with comprehensive test coverage, consistent refactoring patterns across all services, and proper handling of edge cases (empty results, limit=-1, zero values). The changes are well-structured and follow existing architectural patterns.
- No files require special attention
</details>

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| backend/internal/utils/pagination/types.go | Added BuildResponseFromFilterResult helper to handle limit=-1 for "show all" mode |
| backend/internal/utils/pagination/reflect.go | Added paginateDBAll function to bypass pagination when limit=-1 |
| backend/internal/services/container_service.go | Refactored pagination config to use BuildResponseFromFilterResult, added update status filtering |
| frontend/src/lib/components/arcane-table/arcane-table-pagination.svelte | Added "All" option to page size selector that sends limit=-1 to backend |
| frontend/src/lib/components/arcane-table/arcane-table-desktop-view.svelte | Major refactor adding grouping UI with collapsible groups, bulk selection, and sticky columns |
| frontend/src/routes/(app)/projects/components/ProjectContainersTable.svelte | New component for project-specific container table with bulk actions and service management |

</details>

</details>

### CODE REVIEW

File: frontend/src/routes/(app)/containers/container-table.svelte [hunk 1]
```diff
@@ -631,55 +623,28 @@
...
+				{getActionStatusMessage(status)}
+			</span>
 		</div>
-	</div>
+	{:else}
+		<StatusBadge variant={getStateBadgeVariant(item.state)} text={capitalizeFirstLetter(item.state)} />
+	{/if}
+{/snippet}
+
+{#snippet UpdatesCell({ item }: { item: ContainerSummaryDto })}
```
Comment (kmendell) [inline frontend/src/routes/(app)/containers/container-table.svelte:367]: I dont like this, this is unintuitve, the update cell should keep iots existsing behavior, its was all contained in its own thing,

File: frontend/src/routes/(app)/containers/container-table.svelte [hunk 2]
```diff
@@ -801,88 +880,101 @@
 
 {#snippet RowActions({ item }: { item: ContainerSummaryDto })}
```
Comment (kmendell) [inline frontend/src/routes/(app)/containers/container-table.svelte:528]: Im gonna be blunt i think this is too much and ugly. I dont think moving every action to this area is at all a good idea. 

I get what you were going for but i dont like this.

**Comment by cabaucom376** [comment]
So what you're suggesting is instead of inline controls just make them back into a dropdown but sticky?

**Comment by kmendell** [comment]
> So what you're suggesting is instead of inline controls just make them back into a dropdown but sticky?

ultimately yes, So sort of what you had but they should stay in the dropdown, and the inline start and stop and updates stuff should stay where they are as those were accessible enough i thin k?

**Comment by kmendell** [comment]
image

**Comment by kmendell** [comment]
@greptileai

**Review by kmendell — APPROVED** [review]
